python/pyarrow/tests/test_compute.py

```diff
@@ -1755,6 +1755,20 @@ def test_logical():
     assert pc.invert(a) == pa.array([False, True, True, None])
 
 
+def test_dictionary_decode():
+    array = pa.array(["a", "a", "b", "c", "b"])
+    dictionary_array = array.dictionary_encode()
+    dictionary_array_decode = dictionary_array.dictionary_decode()
+
+    assert array != dictionary_array
+
+    assert array == dictionary_array_decode
+    assert array == pc.dictionary_decode(dictionary_array)
+
+    with pytest.raises(TypeError):
+        pc.dictionary_decode(array)
```

Comment:
Shouldn't this be ok now?

Reply:
Good Catch. I also check the input type in python code, let me remove the check